extensions/ql-vscode/test/pure-tests/sarif-utils.test.ts

```diff
@@ -49,6 +49,10 @@ describe('parsing sarif', () => {
       .to.eq('file:C:/a/b/?x=test');
     expect(getPathRelativeToSourceLocationPrefix('C:\\a\\b', '%3Fx%3Dtest'))
       .to.eq('file:C:/a/b/%3Fx%3Dtest');
```

Comment:
These tests are wrong in that file uri paths must always start with a `/`.

Reply:
The first argument to `getPathRelativeToSourceLocationPrefix` is not a URI component. It is a file path that comes from the sarif file. Or at least that is my understanding of how sarif works (and how this extension has been implemented since I arrived).

When I look at sarif files, all the locations are json blobs like this:

```
"location" : {
        "uri" : "include/virtio.h",
        "uriBaseId" : "%SRCROOT%",
        "index" : 1
}
```

The `uri` properties seem to be relative paths.

It's possible that we don't need to do the check for the drive letter (though, not sure about that). Also, maybe should add some more documentation here.